mitodl/open-discussions
# Switch frontend over to user list items API

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2464 

#### What's this PR do?
Generally, this PR adds support for pagination on the lists and favorites APIs, but this required a lot in terms of specifics due to a lot of usages of `items`:

- Added `item_count` to the user list serializer
- Updated user list serializer to return the image of the first item since we can no longer look that up on the frontend

#### How should this be manually tested?

You should be able to do the following:

- add/remove items to/from lists/paths
- open the drawer for user lists
- page through lists in the drawer
- view user lists / paths
- reorder learning paths

Nothing should break

#### Screenshots (if appropriate)

@pdpinch @Ferdi @abdulkdawson :
![Screenshot_2020-01-08 MIT Open Learning](https://user-images.githubusercontent.com/28598/72020409-bd610c00-3239-11ea-9143-50e41d05371f.png)
